src/provider/common/deferredPromise.ts

```diff
@@ -2,7 +2,7 @@
  * Creates a deferred promise and returns it along with handlers to resolve/reject it imperatively
  * @returns a tuple with the promise and its resolve/reject handlers
  */
-export function deferredPromise<T = void>(): readonly [Promise<T>, (value?: T) => void, (reason?: any) => void] {
+export function deferredPromise<T = void>(): DeferredPromise<T> {
```

Comment:
There was a change at some point (probably in a another repo) to make this a proper class. Since we're changing this anyway, it might be worth just dragging that change in here too.

Reply:
Agreed. I'll get it moved over to the class

Reply:
Moved to the class. Much better!

Reply:
Nice!